### PR TITLE
SW-18957 - Update customer stream devdocs, add API endpoint documenta…

### PR DESCRIPTION
**Pull request by jenskueper**

### CODE REVIEW

File: source/developers-guide/customer-streams-extension/index.md [hunk 1]
```diff
@@ -9,11 +9,52 @@ menu_title: Customer - Search & Streams
...
+allow you to create a subgroup of customers by using a combination of over 20
+filters.
+Customer Streams can be used for:
+ 
+* Newsletters, to define the recipients
+* Shopping worlds, to only display a shopping world if the user is part of the
+specified Customer Stream
+* Vouchers
+* Export
+* Advanced Promotions Suite (APS)
```
Comment (mitelg) [inline source/developers-guide/customer-streams-extension/index.md]: the abbreviation is not necessary, I think

File: source/developers-guide/customer-streams-extension/index.md [hunk 2]
```diff
@@ -9,11 +9,52 @@ menu_title: Customer - Search & Streams
...
+
+## Introduction
+In order to create a new Customer Stream, all customers need to be analysed
+before. This process is required to assure high performance.
```
Comment (mitelg) [inline source/developers-guide/customer-streams-extension/index.md]: maybe "ensure high performance"
but I think both is ok

File: source/developers-guide/customer-streams-extension/index.md [hunk 3]
```diff
@@ -9,11 +9,52 @@ menu_title: Customer - Search & Streams
...
+During the analysis, all data which is required for the filters is collected and 
+saved into the `s_customer_search_index` database table.
+
+If you create a new Customer Stream based on some filters, the module will
+check which of the previously analysed customers match these filter and add
```
Comment (mitelg) [inline source/developers-guide/customer-streams-extension/index.md]: ... match these filters and adds ...

File: source/developers-guide/customer-streams-extension/index.md [hunk 4]
```diff
@@ -9,11 +9,52 @@ menu_title: Customer - Search & Streams
...
+them to your Customer Stream.
+
+The mapping between customers and Customer Streams is saved
+in the `s_customer_streams_mapping` table. The stream definition
+(e.g. name, filters) is saved in the database table`s_customer_streams`.
```
Comment (mitelg) [inline source/developers-guide/customer-streams-extension/index.md]: missing space after table

File: source/developers-guide/customer-streams-extension/index.md [hunk 5]
```diff
@@ -9,11 +9,52 @@ menu_title: Customer - Search & Streams
...
+All theses changes are not reflected immediately since all the Customer Stream
+data is basically cached. Therefore you can not rely on a Customer Stream if you
+need to be sure that the data is still up to date.
+
 
 ## Add own condition
 
-Since the customer search is based on the same functions as the product search, the way of extending the search is pretty similar. The customer search is given a criteria object in which the different search conditions are summarized. The first step is to define your own condition:
+Since the customer search is based on the same functions as the product search,
+the way of extending the search is pretty similar. The customer search is given
```
Comment (mitelg) [inline source/developers-guide/customer-streams-extension/index.md]: The customer search gets a criteria object

File: source/developers-guide/customer-streams-extension/index.md [hunk 6]
```diff
@@ -9,11 +9,52 @@ menu_title: Customer - Search & Streams
 menu_order: 500
 ---
 
-This article describes how to extend the customer stream and customer search module in shopware. In most cases, the purpose of a plugin is to provide more conditions or to add data to the search. You find simple examples which describe how this is archived below.
+This article explains how Customer Streams work and how to extend the
+customer stream and customer search module in Shopware. In most cases,
+the purpose of a plugin is to provide more conditions or add additional data to
+the search. You find examples which describe how this is archived below.
```
Comment (jenskueper) [inline source/developers-guide/customer-streams-extension/index.md]: Fixed

File: source/developers-guide/customer-streams-extension/index.md [hunk 7]
```diff
@@ -9,11 +9,52 @@ menu_title: Customer - Search & Streams
...
+
+Shopware 5.3 introduced a new feature called Customer Streams.
+Customer Streams are similar to Product Streams and
+allow you to create a subgroup of customers by using a combination of over 20
+filters.
```
Comment (jenskueper) [inline source/developers-guide/customer-streams-extension/index.md]: Fixed

File: source/developers-guide/customer-streams-extension/index.md (hunk 1 above)
Comment (jenskueper) [inline source/developers-guide/customer-streams-extension/index.md]: Fixed

File: source/developers-guide/customer-streams-extension/index.md [hunk 8]
```diff
@@ -9,11 +9,52 @@ menu_title: Customer - Search & Streams
...
+Customer Streams can be used for:
+ 
+* Newsletters, to define the recipients
+* Shopping worlds, to only display a shopping world if the user is part of the
+specified Customer Stream
+* Vouchers
+* Export
```
Comment (jenskueper) [inline source/developers-guide/customer-streams-extension/index.md]: Fixed

File: source/developers-guide/customer-streams-extension/index.md [hunk 9]
```diff
@@ -9,11 +9,52 @@ menu_title: Customer - Search & Streams
...
+the search. You find examples which describe how this is archived below.
+
+Shopware 5.3 introduced a new feature called Customer Streams.
+Customer Streams are similar to Product Streams and
+allow you to create a subgroup of customers by using a combination of over 20
+filters.
+Customer Streams can be used for:
+ 
+* Newsletters, to define the recipients
+* Shopping worlds, to only display a shopping world if the user is part of the
```
Comment (jenskueper) [inline source/developers-guide/customer-streams-extension/index.md]: Fixed

File: source/developers-guide/customer-streams-extension/index.md [hunk 10]
```diff
@@ -9,11 +9,52 @@ menu_title: Customer - Search & Streams
...
+the purpose of a plugin is to provide more conditions or add additional data to
+the search. You find examples which describe how this is archived below.
+
+Shopware 5.3 introduced a new feature called Customer Streams.
+Customer Streams are similar to Product Streams and
+allow you to create a subgroup of customers by using a combination of over 20
+filters.
+Customer Streams can be used for:
+ 
+* Newsletters, to define the recipients
```
Comment (jenskueper) [inline source/developers-guide/customer-streams-extension/index.md]: Fixed

File: source/developers-guide/customer-streams-extension/index.md (hunk 2 above)
Comment (jenskueper) [inline source/developers-guide/customer-streams-extension/index.md]: Fixed

File: source/developers-guide/customer-streams-extension/index.md (hunk 3 above)
Comment (jenskueper) [inline source/developers-guide/customer-streams-extension/index.md]: Fixed

File: source/developers-guide/customer-streams-extension/index.md (hunk 4 above)
Comment (jenskueper) [inline source/developers-guide/customer-streams-extension/index.md]: Fixed

File: source/developers-guide/customer-streams-extension/index.md (hunk 5 above)
Comment (jenskueper) [inline source/developers-guide/customer-streams-extension/index.md]: Fixed

**Pull request merged**